describe/DOType.ts

```diff
@@ -0,0 +1,122 @@
+import { DADescription, describeDA } from "./DADescription.js";
+import { NamingDescription, describeNaming } from "./Naming.js";
+import { SDODescription, describeSDO } from "./SDODescription.js";
+
+function sortRecord(object: Record<string, any>) {
+  return Object.keys(object)
+    .sort()
+    .reduce((sortedRecord: Record<string, any>, key) => {
+      sortedRecord[key] = object[key];
+
+      return sortedRecord;
+    }, {});
+}
+
+export function isDOTypeDescription(
+  type: DOTypeDescription
+): type is DOTypeDescription {
+  return (type as DOTypeDescription).cdc !== undefined;
+}
+
+type CDCs =
+  | "SPS"
+  | "DPS"
+  | "INS"
+  | "ENS"
+  | "ACT"
+  | "ACD"
+  | "SEC"
+  | "BCR"
+  | "HST"
+  | "VSS"
+  | "MV"
+  | "CMV"
+  | "SAV"
+  | "WYE"
+  | "DEL"
+  | "SEQ"
+  | "HMV"
+  | "HWYE"
+  | "HDEL"
+  | "SPC"
+  | "DPC"
+  | "INC"
+  | "ENC"
+  | "BSC"
+  | "ISC"
+  | "APC"
+  | "BAC"
+  | "SPG"
+  | "ING"
+  | "ENG"
+  | "ORG"
+  | "TSG"
+  | "CUG"
+  | "VSG"
+  | "ASG"
+  | "CURVE"
+  | "CSG"
+  | "DPL"
+  | "LPL"
+  | "CSD"
+  | "CST"
+  | "BTS"
+  | "UTS"
+  | "LTS"
+  | "GTS"
+  | "MTS"
+  | "NTS"
+  | "STS"
+  | "CTS"
+  | "OTS"
+  | "VSD"
+  | "ORS"
+  | "TCS";
+
+export interface DOTypeDescription extends NamingDescription {
+  /** Required attribute cdc */
+  cdc: CDCs;
+  /** Optional attribute iedType*/
+  iedType?: string;
+  /** Child elements DA key is required name attribute of DA child */
+  das: Record<string, DADescription>;
+  /** Child elements SDO key is required name attribute of DA child */
+  sdos: Record<string, SDODescription>;
+}
+
+export function DOType(element: Element): DOTypeDescription {
+  const doTypeDescription: DOTypeDescription = {
+    ...describeNaming(element),
+    cdc: element.getAttribute("cdc")! as CDCs,
```

Comment:
I not have seen any file where this was missing. It does not make sense. HOWEVER, what to do with male formed SCL files? I would not want to describe a function returning an undefined or empty object, as this can lead to false positive compares. 

What other option is there. Do we want to throw an error?

Reply:
In general, throwing an arrow seems preferable to me to returning a seemingly correct description of a malformed SCL element. This would mean try/catch blocks all over the place, so it might be easier on the eyes to simply return `undefined` from description functions called with malformed arguments and then filter that out when not needed.

Solutions I currently see:
* any malformed element anywhere in the hierarchy results in a malformed description (current solution)
* malformed elements are ignored 
  * by the caller, which makes sure all dependencies that it calls "describe" functions on are well-formed before calling "describe" on them and including them in the description
  * by the callee throwing an error and the caller catching it and not adding a dependency if the error is thrown
  * by the callee returning undefined and the caller filtering out undefined results and not adding a dependency
* any malformed element anywhere in the hierarchy results in no description at all, instead an error is thrown by the function attempting to describe the malformed element, never caught anywhere, and propagated up to the original call site.

Do you see additional solutions? Which of them sounds best to you?

Reply:
I would go with returning `undefined` and the callee filtering this when adding in other describe functions.